src/model/RatePair.ts

```diff
@@ -0,0 +1,29 @@
+import { RateProvider } from "@/model/RateProvider";
+
+export class RatePair {
+  private rateProvider: RateProvider;
+  private readonly _codeFrom: string;
+  private readonly _codeTo: string;
+
+  constructor(codeFrom: string, codeTo: string, rateProvider: RateProvider) {
+    this.rateProvider = rateProvider;
+    this._codeFrom = codeFrom;
+    this._codeTo = codeTo;
+    this.updateRate();
```

Comment:
Maybe this makes it more difficult to test the class

Reply:
Great point! Fixed.